capstone/src/main/webapp/visualizationdata.js

```diff
@@ -0,0 +1,215 @@
+// limitations under the License.
+
+/**
+ * Gets the page view data using the PageViewServlet.
+ */
+async function getPageViewData() {
+  // Fetch data from servlet
+  const responsePath = '/page-view';
+  const response = await fetch(responsePath);
+  return await response.json();
+}
+
+/**
+ * Parses the page view data into the desired format for the
+ * multi-line chart visualization.
+ * @return data object of the form:
+ * {
+ *     restaurantData: [
+ *         {
+ *             restaurantName: String,
+ *             clickData: [int(# page views that week), int, int, ...], lines up
+ * with dates array
+ *         },
+ *         ...
+ *     ],
+ *     dates: [Date, Date(one week later), ..., Date(this week)],
+ * }
+ */
+async function parseData() {
+  const restaurantPageViews = await getPageViewData();
+
+  const firstDate = getFirstDate(restaurantPageViews);
+  if (firstDate.length === 0) {
+    // restaurantPageViews was empty: no data for d3 chart
+    return [];
+  }
+  const firstDateObj = getDateFromWeekYear(firstDate[0], firstDate[1]);
+  const currDate = new Date();
+  const dateArray = getFullDateArray(firstDateObj, currDate);
+
+  return setUpVisualizationData(restaurantPageViews, dateArray);
+}
+
+/**
+ * Gets the earliest date with data based on the pageViews array.
+ * @param restaurantPageViews
+ * @return an array containing minWeek and minYear, or an empty array if no data
+ *     was supplied
+ */
+function getFirstDate(restaurantPageViews) {
+  // Check for if there is no data stored yet
+  if (restaurantPageViews == null || restaurantPageViews.length <= 0) {
+    return [];
+  }
+
+  // Start the variables at the first data point values
+  // Assumes any populated restaurantPageViews member will have at least one
+  // element in its pageView array, which is guaranteed by the backend
+  let minWeek = restaurantPageViews[0].pageViews[0].week;
+  let minYear = restaurantPageViews[0].pageViews[0].year;
+
+  // Check through rest of the first data points in other restaurants'
+  // pageViews arrays to see if any of the dates are earlier
+  // Assumes each RestaurantPageView pageViews member is sorted in ascending
+  // order, which is guaranteed by the backend
+  for (let i = 1; i < restaurantPageViews.length; i++) {
+    const currRestaurantMinPageView = restaurantPageViews[i].pageViews[0];
+    if (currRestaurantMinPageView.year < minYear) {
+      minYear = currRestaurantMinPageView.year;
+      minWeek = currRestaurantMinPageView.week;
+    } else if (
+        currRestaurantMinPageView.week < minWeek &&
+        currRestaurantMinPageView.year === minYear) {
+      minWeek = currRestaurantMinPageView.week;
+    }
+  }
+
+  return [minWeek, minYear];
+}
+
+/**
+ * Get a javascript Date object based on a week and year
+ * @param week
+ * @param year
+ * @return {Date}
+ */
+function getDateFromWeekYear(week, year) {
+  if (week < 0 || year < 0) {
+    return null;
+  }
+  // January 1st plus 7 days for each week
```

Comment:
Good idea, I added it there. I also made sure the days returned will be Sundays now because that works best for the visualization; added that to the method documentation as well.